genie-web/src/main/java/com/netflix/genie/web/services/JobResolverService.java

```diff
@@ -54,8 +54,9 @@
      *
      * @param id         The id of the job
      * @param jobRequest The job request containing all details a user wants to have for their job
+     * @param apiJob     {@literal true} if this job was submitted via the REST API. {@literal false} otherwise.
      * @return The completely resolved job information within a {@link ResolvedJob} instance
      */
     @Nonnull
-    ResolvedJob resolveJob(String id, @Valid JobRequest jobRequest);
+    ResolvedJob resolveJob(String id, @Valid JobRequest jobRequest, boolean apiJob);
```

Comment:
Seems strange we have to pass this boolean in order to resolve... It doesn't actually influence the resolution so seems off to have it here

Reply:
This `resolveJob` signature is the three argument one. It is used in three places (currently). The dry run job specification, the resolve job specification call by the agent (will no longer be the case once my other work for agent execution is incorporated) and within the `JobCoordinatorServiceImpl` which is the V3 code path. 2 of these 3 are non-api but the other is only API. There was no field in the `JobRequest` which captured this data as it is in `JobRequestMetadata` as it's more data about the request then something actually provided by the user (which is what's in the job request DTO)

It is required/influences the resolution because of the cascading rules we discussed which is implemented below:
```
        final Integer timeout;
        if (jobRequest.getRequestedAgentConfig().getTimeoutRequested().isPresent()) {
            timeout = jobRequest.getRequestedAgentConfig().getTimeoutRequested().get();
        } else if (apiJob) {
            // For backwards V3 compatibility
            timeout = com.netflix.genie.common.dto.JobRequest.DEFAULT_TIMEOUT_DURATION;
        } else {
            timeout = null;
        }
```